framework/src/main/java/org/checkerframework/framework/flow/CFAbstractStore.java

```diff
@@ -217,19 +286,55 @@ public void updateForMethodCall(
             // TODO: Also remove if any element/argument to the annotation is not
             // isUnmodifiableByOtherCode.  Example: @KeyFor("valueThatCanBeMutated").
             if (sideEffectsUnrefineAliases) {
-                localVariableValues
-                        .entrySet()
-                        .removeIf(e -> !e.getKey().isUnmodifiableByOtherCode());
+                if (sideEffectExpressions != null) {
+                    MethodAccessNode methodTarget = n.getTarget();
+                    Node receiver = methodTarget.getReceiver();
+
+                    final List<String> expressionsToRemove = sideEffectExpressions;
+                    localVariableValues
+                            .entrySet()
+                            .removeIf(
+                                    e ->
+                                            (expressionsToRemove.contains(e.getKey().toString())
+                                                            && !e.getKey()
```

Comment:
I think it would be clearer (and the code would be shorter) if this subexpression appeared just once rather than twice.